src/scripts/main.js

```diff
@@ -1,3 +1,98 @@
 'use strict';
 
-// write code here
+const field = document.querySelector('.container');
+const tBody = document.querySelector('tbody');
+const addRowBtn = document.querySelector('.append-row');
+const removeRowBtn = document.querySelector('.remove-row');
+const addColBtn = document.querySelector('.append-column');
+const removeColBtn = document.querySelector('.remove-column');
+
+function addRow(table) {
+  table.lastElementChild.after(table.lastElementChild.cloneNode(true));
+}
+
+function removeRow(table) {
+  table.lastElementChild.remove();
+}
+
+function addColumn(table) {
+  Array.from(table.rows).forEach(row => row
+    .lastElementChild
+    .after(row.lastElementChild.cloneNode(true))
+  );
+}
+
+function removeColumn(table) {
+  Array.from(table.rows).forEach(row => row
+    .lastElementChild.remove()
+  );
+}
+
+function setAttrDisabled(el) {
+  el.setAttribute('disabled', '');
+}
+
+function removeAttrDisabled(el) {
+  el.removeAttribute('disabled');
+}
+
+field.addEventListener('click', (ev) => {
+  const buttons = ev.target.closest('.button');
+  const lengthRows = tBody.rows.length;
+  const lengthCells = tBody.rows[0].cells.length;
+
+  if (!buttons) {
+    return;
+  }
+
+  const maxLengthRows = 9;
+  const minLengthRows = 3;
+  const maxLengthCol = 9;
+  const minLengthCol = 3;
+
+  switch (buttons) {
+    case addRowBtn:
+      if (lengthRows <= maxLengthRows) {
+        addRow(tBody);
+        removeAttrDisabled(removeRowBtn);
+      }
+
+      if (lengthRows === maxLengthRows) {
+        setAttrDisabled(addRowBtn);
+      }
+      break;
+
+    case removeRowBtn:
+      if (lengthRows >= minLengthRows) {
+        removeRow(tBody);
+        removeAttrDisabled(addRowBtn);
+      }
+
+      if (lengthRows <= minLengthRows) {
+        setAttrDisabled(removeRowBtn);
+      }
+      break;
+
+    case addColBtn:
+      if (lengthCells <= maxLengthCol) {
+        addColumn(tBody);
+        removeAttrDisabled(removeColBtn);
+      }
+
+      if (lengthCells >= maxLengthCol) {
+        setAttrDisabled(addColBtn);
+      }
```

Comment:
```suggestion
      if (lengthCells <= maxLengthCol) {
        addColumn(tBody);
        removeAttrDisabled(removeColBtn);
      } else {
        setAttrDisabled(addColBtn);
      }
```

Reply:
I agree. It works. But when count column =10 button don` set disable on button. Set disable after click else one. Is it right?
<img width="678" alt="Screenshot 2022-12-21 at 10 33 55" src="https://user-images.githubusercontent.com/102901051/208857770-68c3485b-7053-44c6-8e85-9420d4d57b03.png">

Reply:
You can check if lengthCells < (instead of <=) maxLengthCol, and it should work right on else then